Library/OcAppleKernelLib/MkextContext.c

```diff
@@ -0,0 +1,1036 @@
+/** @file
+  Mkext support.
+
+  Copyright (c) 2020, Goldfish64. All rights reserved.
+
+  This program and the accompanying materials
+  are licensed and made available under the terms and conditions of the BSD License
+  which accompanies this distribution.  The full text of the license may be found at
+  http://opensource.org/licenses/bsd-license.php
+
+  THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+  WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+
+**/
+
+#include <Uefi.h>
+
+#include <IndustryStandard/AppleFatBinaryImage.h>
+
+
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/OcAppleKernelLib.h>
+#include <Library/OcCompressionLib.h>
+#include <Library/OcFileLib.h>
+#include <Library/OcGuardLib.h>
+#include <Library/OcStringLib.h>
+
+#define MKEXT_OFFSET_STR_LEN    24
+
+//
+// Alignment to 8 bytes.
+//
+#define MKEXT_ALIGN(a)         (ALIGN_VALUE (a, sizeof (UINT64)))
+
+STATIC
+BOOLEAN
+ParseKextBinary (
+  IN OUT UINT8         **Buffer,
+  IN OUT UINT32        *BufferSize,
+  IN     BOOLEAN       Is64Bit
+  )
+{
+  MACH_HEADER_ANY *MachHeader;
+  
+  MachoFilterFatArchitectureByType (Buffer, BufferSize, Is64Bit ? MachCpuTypeX8664 : MachCpuTypeI386);
+  MachHeader = (MACH_HEADER_ANY *)* Buffer; // TODO alignment?
+
+  if ((!Is64Bit && MachHeader->Signature == MACH_HEADER_SIGNATURE)
```

Comment:
Is it guaranteed to be little endian? I once remember seeing big endian i386 mach-o.